components/Common/ThemeToggle/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { MoonIcon, SunIcon } from '@heroicons/react/24/outline';
+import { AccessibleIcon } from '@radix-ui/react-accessible-icon';
+import { useTheme } from 'next-themes';
+import type { FC } from 'react';
+import { useIntl } from 'react-intl';
+
+import styles from './index.module.css';
+
+const ThemeToggle: FC = () => {
+  const intl = useIntl();
+  const { theme, setTheme } = useTheme();
+
+  const isDark = theme === 'dark';
+
+  const toggleTheme = (isKeyPress?: boolean) => {
+    if (isKeyPress) {
+      return;
+    }
+
+    setTheme(isDark ? 'light' : 'dark');
+  };
+
+  return (
+    <button
+      type="button"
+      onClick={() => toggleTheme()}
+      onKeyDown={() => toggleTheme(true)}
+    >
+      <AccessibleIcon
+        label={intl.formatMessage({
+          id: 'components.header.buttons.toggleDarkMode',
+        })}
+      >
+        {isDark ? (
+          <SunIcon className={styles.icon} />
```

Comment:
I think the rendering of these icons should be done via CSS, in the sense that we should always render both but hide the one we're not showing.

```tsx
<SunIcon className="dark:hidden" />
<MoonIcon className="dark:block hiddne" />
```

Also, the styling shouldn't be done in the icon `styles.icon` but on the container. And then use something like:


```css
.className svg {}
```

for the actual SVG styles

Reply:
I agree with you that it would be simpler